### PR TITLE
Add test-configuration.json documentation

### PR DESCRIPTION
**Pull request by ChadNedzlek**

### CODE REVIEW

File: src/Microsoft.DotNet.Helix/Sdk/Readme.md [hunk 1]
```diff
@@ -286,3 +286,80 @@ You may assume that all the following variables are set on any given Helix clien
...
+## localRerunCount
+- default: 1
+
+This number indicates the number of times a test that needs to be "rerun" should be rerun on the local computer immediately.
+This is the fastest rerun option, because the payloads don't need to be redownloaded, so it always the first attempted re-execution method.
+
+In the example, with a value of "2", that means that the test will need to fail 3 times before being marks as failed (1 intial failure, and 2 rerun failures).
+
+## rules
+The three "rules" entries are lists of rules that will be used to match test to determin desired behavior. In the case of multiple rule matches:
```
Comment (danmoseley) [inline src/Microsoft.DotNet.Helix/Sdk/Readme.md:338]:
```suggestion
The three "rules" entries are lists of rules that will be used to match test to determine desired behavior. In the case of multiple rule matches:
```

File: src/Microsoft.DotNet.Helix/Sdk/Readme.md [hunk 2]
```diff
@@ -286,3 +286,80 @@ You may assume that all the following variables are set on any given Helix clien
...
+
+All comparisons are case-insensitive
+#### Raw string (e.g. "rule string")
+True if the property value exactly matches the string
+#### {"contains": "value"}
+True if the property contains (case-insensitive) the value string
+#### {"wildcard": "value with * wildcard"}
+The same as a raw string, but "*" can match any number of characters, and "?" can match one character
+#### {"regex": "value with .* regex"}
+true if the propety matches the regular expression
```
Comment (danmoseley) [inline src/Microsoft.DotNet.Helix/Sdk/Readme.md:365]:
```suggestion
true if the property matches the regular expression
```

File: src/Microsoft.DotNet.Helix/Sdk/Readme.md [hunk 3]
```diff
@@ -286,3 +286,80 @@ You may assume that all the following variables are set on any given Helix clien
...
+This number indicates the number of times a test that needs to be "rerun" should be rerun on the local computer immediately.
+This is the fastest rerun option, because the payloads don't need to be redownloaded, so it always the first attempted re-execution method.
+
+In the example, with a value of "2", that means that the test will need to fail 3 times before being marks as failed (1 intial failure, and 2 rerun failures).
+
+## rules
+The three "rules" entries are lists of rules that will be used to match test to determine desired behavior. In the case of multiple rule matches:
+- if a quarantine rule matches, the test is quarantined
+- if the default behavior is "rerun" and a "fail" rule matches, the test is failed
+- if the default bheavior is "fail" and a "rerun" rule matches, the test is rerun
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.Helix/Sdk/Readme.md:341]:
```suggestion
- if the default behavior is "fail" and a "rerun" rule matches, the test is rerun
```

**Comment by ChadNedzlek** [comment]
Could one of @AlitzelMendez  or @garath sign off on this guy?

File: src/Microsoft.DotNet.Helix/Sdk/Readme.md [hunk 4]
```diff
@@ -286,3 +286,80 @@ You may assume that all the following variables are set on any given Helix clien
...
 
 
+## Test Retry
+Helix supports retrying and reporting on partial successes for tests based on repository specific configuration.
+When the configuration matches a test failure, the test assembly is reexecuted, and the results compared.
+Tests that failed partially (only in some executions) will be reported to Azure DevOps as "Passed on Rerun",
+which will include each iteration as a sub-result of the failing test.
+If a test passes or fails in all attempts, only a single report is made representing the first execution.
+
+To opt-in and configure test retries when using helix, create file in the reporitory at "eng/test-configuration.json"
```
Comment (garath) [inline src/Microsoft.DotNet.Helix/Sdk/Readme.md:296]:
```suggestion
To opt-in and configure test retries when using helix, create a file in the reporitory at "eng/test-configuration.json"
```

**Review by garath — APPROVED** [review]

**Pull request merged**